README.md

```diff
@@ -1,29 +1,310 @@
+
+#[tokio::test]
+async fn example() {
+    // Acquire an OpenFeature API instance.
+    // Note the `await` call here because asynchronous lock is used to guarantee thread safety.
+    let mut api = OpenFeature::singleton_mut().await;
+
+    // Create an unnamed client.
+    let client = api.create_client();
+
+    // Create an evaluation context.
+    // It supports types mentioned in the specification.
+    //
+    // You have multiple ways to add a custom field.
+    let evaluation_context = EvaluationContext::builder()
+        .targeting_key("Targeting")
+        .build()
+        .with_custom_field("bool_key", true)
+        .with_custom_field("int_key", 100)
+        .with_custom_field("float_key", 3.14)
+        .with_custom_field("string_key", "Hello".to_string())
+        .with_custom_field("datetime_key", time::OffsetDateTime::now_utc())
+        .with_custom_field(
+            "struct_key",
+            EvaluationContextFieldValue::Struct(Arc::new(MyStruct::default())),
+        )
+        .with_custom_field("another_struct_key", Arc::new(MyStruct::default()))
+        .with_custom_field(
+            "yet_another_struct_key",
+            EvaluationContextFieldValue::new_struct(MyStruct::default()),
+        );
+
+    assert_eq!(
+        client
+            .get_bool_value("key", Some(&evaluation_context), None)
+            .await
+            .unwrap(),
+        bool::default()
+    );
+
+    // Let's get evaluation details.
+    // Note that we will inject `300` as the int value via evaluation context.
+    // It is not a feature mentioned in the standard but rather implemented for the
+    // convenience.
+    let result = client
+        .get_int_details(
+            "key",
+            Some(&EvaluationContext::default().with_custom_field("Value", 300)),
+            None,
+        )
+        .await;
+
+    match result {
+        Ok(details) => {
+            assert_eq!(details.value, 300);
+            assert_eq!(details.reason, Some(EvaluationReason::Static));
+            assert_eq!(details.variant, Some("Static".to_string()));
+            assert_eq!(details.flag_metadata.values.iter().count(), 1);
+        }
+        Err(error) => {
+            println!(
+                "Error: {}\nMessage: {:?}\n",
+                error.code.to_string(),
+                error.message
+            );
+        }
+    }
+}
+```
+
+#### Getting a Struct from a Provider
+
+It is possible to extract a struct from the provider. Internally, this SDK defines a type `StructValue` to store any structure value. The `client.get_struct_value()` functions takes a type parameter `T`. It will try to parse `StructValue` resolved by the provider to `T`, as long as `T` implements trait `FromStructValue`.
+
+You can pass in a type that satisfies this trait bound. When the conversion fails, it returns an `Err` with `EvaluationReason::TypeMismatch`.
+
+### API Reference
+
+<!-- TODO: link to formal API docs (ie: Javadoc) if available -->
```

Comment:
Does Rust have a way to generate API docs?

Reply:
Yes. There is an official https://docs.rs that holds the document of a crate (package). I think it is automatically done when the crates is pushed to https://crates.io.

I will do that after this PR is merged and publish it to crates.io.

Reply:
It extracts comments starting with `///` and display them as Markdown. I will come back and update README when the docs are there. :-)